run_octane.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * QuickJS Javascript Engine Benchmark Suite
+ *
+ * Copyright (c) 2023 Divy Srivastava
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "quickjs/quickjs.h"
+#include "quickjs/quickjs-libc.h"
+
+void Execute(JSContext *ctx, const char *filename, int eval_flags) {
+    FILE *file = fopen(filename, "r");
+    /* assumes max test script size */
+    static char buf[8 << 20]; // 8 MB
+    if (file == NULL) {
+        fprintf(stderr, "Cannot open file: %s\n", filename);
+        return;
+    }
+    size_t nread = fread(buf, 1, sizeof(buf), file);
+    fclose(file);
+    if (nread == 0) {
+        fprintf(stderr, "fread failed: %s\n", filename);
+        return;
+    }
+    if (nread == sizeof(buf)) {
+        fprintf(stderr, "file too large: %s\n", filename);
+        return;
+    }
+    buf[nread] = '\0';
+
+    JSValue ret_val = JS_Eval(ctx, buf, nread, filename, eval_flags);
+    if (JS_IsException(ret_val)) {
+        JSValue exception = JS_GetException(ctx);
+        printf("Error %s\n", JS_ToCString(ctx, exception));
+        JS_FreeValue(ctx, exception);
+    }
+    JS_FreeValue(ctx, ret_val);
+    return;
+}
+
+int main(int argc, char **argv) {
+    JSRuntime *rt = JS_NewRuntime();
+    JSContext *ctx = JS_NewContext(rt);
+
+    /* system modules */
+    js_init_module_std(ctx, "std");
+
+    JS_SetMaxStackSize(rt, 864 * 1024); // 864 KB
```

Comment:
Should the stack checks be disabled for the test? Setting it to zero would accomplish that.